### PR TITLE
feat: add map generator and display

### PR DESCRIPTION
**Pull request by ktos**

### CODE REVIEW

File: Server/Modules/MapGenerator/Services/MapGeneratorService.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    private MapData _mapData;
+
+    private float persistance = 0.5f;
+    private float lacunarity = 2f;
+    private int octaves = 5;
+    private float scale = 5f;
+
+    private Random rnd = new Random();
+
+    //private Dictionary<(TemperatureType, HeightType), Biome> biomesDictionary = new Dictionary<(TemperatureType, HeightType), Biome>
```
Comment (purifetchi) [inline Server/Modules/MapGenerator/Services/MapGeneratorService.cs:19]: unless this code is to be used later, i don't think we should keep commented out code

File: Server/Modules/MapGenerator/Services/MapGeneratorService.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    private MapData GenerateMap(NoiseParameters noiseParameters)
+    {
+        int width = 32;
+        int height = 32;
+        bool useFalloff = true;
+        seed = noiseParameters.UseRandomSeed ? rnd.Next(int.MaxValue) : noiseParameters.Seed;
+
+        Node[,] grid = new Node[width, height];
```
Comment (purifetchi) [inline Server/Modules/MapGenerator/Services/MapGeneratorService.cs:56]: `var` please

File: Server/Modules/NoiseGenerator/Services/Perlin.cs [hunk 3]
```diff
@@ -0,0 +1,396 @@
+﻿namespace Fracture.Server.Modules.NoiseGenerator.Services;
```
Comment (purifetchi) [inline Server/Modules/NoiseGenerator/Services/Perlin.cs:1]: to be fair i think we should sooner or later replace our hand-rolled implementation of the perlin noise generator with a NuGet package

File: Server/Modules/MapGenerator/Services/MapGeneratorService.cs [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                if (useFalloff)
+                {
+                    heightMap[x, y] = Math.Clamp(heightMap[x, y] - falloffMap[x, y], 0, 1);
+                }
+
+                grid[x, y] = new Node(x, y, null);
+                grid[x, y].NoiseValue = heightMap[x, y];
+                grid[x, y].Walkable = grid[x, y].NoiseValue > 0.2f && grid[x, y].NoiseValue < 0.7f;
+
+                if (heightMap[x, y] < 0.02f)
```
Comment (purifetchi) [inline Server/Modules/MapGenerator/Services/MapGeneratorService.cs:92]: i... don't really like this, although i can't come up with a better solution off the top of my head for now.

(i assume this is for biomes?)

**Review by purifetchi — CHANGES_REQUESTED** [review]
some initial style issues, the code seems fine otherwise

the `var` comment applies to most of the files too, but i didn't want to spam every single file with the same comment

**Comment by purifetchi** [comment]
`Program.cs` is conflicting

**Comment by ktos** [comment]
Rebased.

**Review by purifetchi — APPROVED** [review]

**Pull request merged**